src/celeritas/global/CoreTrackData.hh

```diff
@@ -90,13 +91,17 @@ struct CoreStateData
     template<class T>
     using Items = StateCollection<T, W, M>;
 
+    using TrackSlotsLookup = Collection<TrackSlotId::size_type, W, M, ThreadId>;
```

Comment:
This nomenclature is a bit unusual; it's describing what an instance of the type does rather than the type itself. What about:
```suggestion
    template<class T>
    using ThreadItems = Collection<T, W, M, ThreadId>;
```
and below:
```c++
    ThreadItems<TrackSlotId::size_type> track_slots;
```